model/infinispan/src/main/java/org/keycloak/models/sessions/infinispan/util/SessionTimeouts.java

```diff
@@ -103,25 +103,26 @@ public static long getUserSessionMaxIdleMs(RealmModel realm, UserSessionEntity u
      * @return
      */
     public static long getClientSessionLifespanMs(RealmModel realm, AuthenticatedClientSessionEntity clientSessionEntity) {
```

Comment:
Just thinking loud, that if this method also has `UserSessionModel` (or entity) as an argument, it will allow to:
- avoid introducing `USER_SESSION_STARTED_AT_NOTE` (which is another thing saved in the infinispan cache)
- allow more proper computation of `sessionMaxLifespan` with something like `int sessionMaxLifespan = userSession.isRememberMe() ? Math.max(realm.getSsoSessionMaxLifespan(), realm.getSsoSessionMaxLifespanRememberMe()) : realm.getSsoSessionMaxLifespan();`

On the other hand, it seems that having userSession available here may be fair amount of refactoring (and still not 100% sure if it is doable...). Also problem with not-100% accurate `sessionMaxLifespan` was even before this PR... So rather not worth an effort, but we're keeping as is now? Was just thinking if someone feels we should try to go this way, but I am myself rather not very keen on that...;-)

Reply:
Yeah! I also thought about getting the user session somehow. But I detected a big change (probably we would need the user session id in the authentication session) and I decided to follow contributors idea which seems to be easier.

Reply:
yeah, agree. Thanks!

Reply:
Hi, thanks @rmartinc for taking both changes in the fix. However, after I submitted my first PR I realized that to properly calculating the the lifespan we need to know if the user session has 'remember me' on or not, so apart from `USER_SESSION_STARTED_AT_NOTE` I added also note if it's remember me.
I got it @mposolda it's yet another thing to keep but in our specific case with remember me set to 180 days compared to couple hours without it, memory increase with these attribute is nothing with amount of sessions that are deleted thanks to proper calculation.

Reply:
@pjober If we also want to include the remember me then maybe it's better to directly store the user session id. I think it will need a big refactoring but it probably makes sense. @mposolda WDYT? (I suppose this will only be for the infinispan implementation.)

Reply:
@rmartinc @pjbober Will it be possible to have "remember me" as another note on AuthenticatedClientSessionModel? The storing userSessionId instead on AuthenticatedClientSessionModel (instead of "Started at" and "remember me") works as well, but if it requires a big refactoring or additional lookup of userSession from infinispan, then my preference would be to rather stick with "Started at" and "remember me" notes.